state/host.go

```diff
@@ -0,0 +1,39 @@
+package state
+
+import (
+	"errors"
+)
+
+// HostImpl maintain all hosts mapping name <=> host
+type HostImpl struct {
```

Comment:
HostImpl is not a good name.

1. a struct may implement multiple interfaces.
2. If we have two host implementation, what's the second name?

so, here I think MapHost is better.

Maybe we will have MemoryPoolHost, SyncMapHost in future.